File: backend/app/Models/Notif.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Notif extends Model
{
    use HasFactory;

    public $timestamps = false;

    public static array $visibility = [
        'seen',
        'unseen',
    ];


    public function account()
    {
        return $this->belongsTo(Account::class);

    }

    public function lead()
    {
        return $this->belongsTo(Lead::class);

    }

    public function thread()
    {
        return $this->belongsTo(Thread::class);
    }

    public function message()
    {
        return $this->belongsTo(Message::class);
    }
}

File: script/models/Notif.py
from datetime import datetime

from peewee import *
from .BaseWithTimeZoneModel import BaseWithTimeZoneModel
from .Account import Account
from .Lead import Lead
from .Thread import Thread
from .Message import Message


class Notif(BaseWithTimeZoneModel):
    account = ForeignKeyField(Account, backref='notifs', null=True)
    lead = ForeignKeyField(Lead, backref='notifs', null=True)
    thread = ForeignKeyField(Thread, backref='notifs', null=True)
    message = ForeignKeyField(Message, backref='notifs', null=True)
    visibility = CharField(default='unseen')


    class Meta:
        table_name = 'notifs'
